From 50de84a394478cc8d55167532e4ecaf0d6c6ccdf Mon Sep 17 00:00:00 2001
From: JuanMiguelBG <bacojmbg@gmail.com>
Date: Mon, 22 May 2023 19:21:58 +0200
Subject: [PATCH] Screenshot directory

---
 base/commandLine.cpp | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/base/commandLine.cpp b/base/commandLine.cpp
index 20fd0faf..83c12afa 100644
--- a/base/commandLine.cpp
+++ b/base/commandLine.cpp
@@ -153,6 +153,7 @@ static const char HELP_STRING4[] =
 	"  --joystick[=NUM]         Enable joystick input (default: 0 = first joystick)\n"
 	"  --savepath=PATH          Path to where saved games are stored\n"
 	"  --extrapath=PATH         Extra path to additional game data\n"
+	"  --screenshot=PATH        Path to where screenshots are saved\n"
 	"  --iconspath=PATH         Path to additional icons for the launcher grid view\n"
 	"  --soundfont=FILE         Select the SoundFont for MIDI playback (only\n"
 	"                           supported by some MIDI drivers)\n"
@@ -312,6 +313,8 @@ void registerDefaults() {
 
 	ConfMan.registerDefault("enable_unsupported_game_warning", true);
 
+	ConfMan.registerDefault("screenshotspath", ".");
+
 	// Game specific
 	ConfMan.registerDefault("path", "");
 	ConfMan.registerDefault("platform", Common::kPlatformDOS);
@@ -867,6 +870,15 @@ Common::String parseCommandLine(Common::StringMap &settings, int argc, const cha
 			DO_LONG_OPTION("md5-engine")
 			END_OPTION
 
+			DO_LONG_OPTION("screenshotspath")
+				Common::FSNode path(option);
+				if (!path.exists()) {
+					usage("Non-existent screenshots path '%s'", option);
+				} else if (!path.isWritable()) {
+					usage("Non-writable screenshots path '%s'", option);
+				}
+			END_OPTION
+
 			DO_LONG_OPTION_INT("talkspeed")
 			END_OPTION
 
-- 
2.25.1

